src/System.Slices/src/System/Span.cs

```diff
@@ -87,6 +87,8 @@ public Span(T[] array, int start, int length)
         {
             Contract.Requires(array != null);
             Contract.RequiresInInclusiveRange(start, array.Length);
+            Contract.RequiresNonNegative(length);
+            Contract.RequiresInInclusiveRange(start + length, array.Length);
             if (start < array.Length)
```

Comment:
I meant the check inside the method, not the Contract checks. I thought that the contract checks are already verifying the if (start < array.Length). If that's not the case, than I have no issues with it.